### ISSUE TITLE
Elasticdump exports invalid json according to several json linting tools

### ISSUE THREAD

**Issue by pgporada** [taskrabbit/elasticsearch-dump#291]
- ElasticDump version `3.1.0`

- Elasticsearch version `5.2.1-1`

- Full command you are having issue with (with inputs and outputs)
```
export NODE_TLS_REJECT_UNAUTHORIZED=0 
INDICES=$(curl -skXGET https://es.phil/_cat/indices?pretty | cut -d $' ' -f3)
for INDEX in $INDICES; do
    echo $INDEX - mapping
    elasticdump \
        --bulk \
        --type=mapping \
        --input=https://es.phil/${INDEX} \
        --output=$ | gzip > ${INDEX}.mapping.json.gz

    echo $INDEX - data
    elasticdump \
        --bulk \
        --type=data \
        --input=https://es.phil/${INDEX} \
        --output=$ | gzip > ${INDEX}.data.json.gz
done
```

- A data set you can share publicly which reproduces the problem
```
$ cat logstash-2017.02.14.mapping.json 
{"logstash-2017.02.14":{"mappings":{"ansible":{"_all":{"enabled":true,"norms":false},"dynamic_templates":[{"message_field":{"path_match":"message","match_mapping_type":"string","mapping":{"norms":false,"type":"text"}}},{"string_fields":{"match":"*","match_mapping_type":"string","mapping":{"fields":{"keyword":{"type":"keyword"}},"norms":false,"type":"text"}}}],"properties":{"@timestamp":{"type":"date","include_in_all":false},"@version":{"type":"keyword","include_in_all":false},"ansible_changed":{"type":"boolean"},"ansible_facts":{"type":"text","norms":false,"fields":{"keyword":{"type":"keyword"}}},"ansible_host":{"type":"text","norms":false,"fields":{"keyword":{"type":"keyword"}}},"ansible_playbook":{"type":"text","norms":false,"fields":{"keyword":{"type":"keyword"}}},"ansible_result":{"type":"text","norms":false,"fields":{"keyword":{"type":"keyword"}}},"ansible_task":{"type":"text","norms":false,"fields":{"keyword":{"type":"keyword"}}},"ansible_type":{"type":"text","norms":false,"fields":{"keyword":{"type":"keyword"}}},"geoip":{"dynamic":"true","properties":{"ip":{"type":"ip"},"latitude":{"type":"half_float"},"location":{"type":"geo_point"},"longitude":{"type":"half_float"}}},"host":{"type":"text","norms":false,"fields":{"keyword":{"type":"keyword"}}},"level":{"type":"text","norms":false,"fields":{"keyword":{"type":"keyword"}}},"logger_name":{"type":"text","norms":false,"fields":{"keyword":{"type":"keyword"}}},"message":{"type":"text","norms":false},"path":{"type":"text","norms":false,"fields":{"keyword":{"type":"keyword"}}},"port":{"type":"long"},"session":{"type":"text","norms":false,"fields":{"keyword":{"type":"keyword"}}},"status":{"type":"text","norms":false,"fields":{"keyword":{"type":"keyword"}}},"type":{"type":"text","norms":false,"fields":{"keyword":{"type":"keyword"}}}}},"_default_":{"_all":{"enabled":true,"norms":false},"dynamic_templates":[{"message_field":{"path_match":"message","match_mapping_type":"string","mapping":{"norms":false,"type":"text"}}},{"string_fields":{"match":"*","match_mapping_type":"string","mapping":{"fields":{"keyword":{"type":"keyword"}},"norms":false,"type":"text"}}}],"properties":{"@timestamp":{"type":"date","include_in_all":false},"@version":{"type":"keyword","include_in_all":false},"geoip":{"dynamic":"true","properties":{"ip":{"type":"ip"},"latitude":{"type":"half_float"},"location":{"type":"geo_point"},"longitude":{"type":"half_float"}}}}}}}}
```

```
$ cat logstash-2017.02.14.data.json 
{"_index":"logstash-2017.02.14","_type":"ansible","_id":"AVo7OmJkU0QLp7y3Mj6e","_score":1,"_source":{"ansible_type":"start","level":"INFO","session":"ab28360e-f27b-11e6-b8c1-0242ac110004","message":"START /tmp/kitchen/default.yml","type":"ansible","ansible_playbook":"/tmp/kitchen/default.yml","tags":[],"path":"/tmp/kitchen/callback_plugins/logstash.py","@timestamp":"2017-02-14T06:06:04.786Z","port":49716,"@version":"1","host":"3c80185cc093","logger_name":"python-logstash-logger","status":"OK"}}
{"_index":"logstash-2017.02.14","_type":"ansible","_id":"AVo7PIMBvGV8lhZBQyFO","_score":1,"_source":{"ansible_type":"start","level":"INFO","session":"ff0b955e-f27b-11e6-822b-0242ac110004","message":"START /tmp/kitchen/default.yml","type":"ansible","ansible_playbook":"/tmp/kitchen/default.yml","tags":[],"path":"/tmp/kitchen/callback_plugins/logstash.py","@timestamp":"2017-02-14T06:08:25.526Z","port":49750,"@version":"1","host":"3c80185cc093","logger_name":"python-logstash-logger","status":"OK"}}
{"_index":"logstash-2017.02.14","_type":"ansible","_id":"AVo7PI0nvGV8lhZBQyFR","_score":1,"_source":{"ansible_type":"task","level":"INFO","session":"ff0b955e-f27b-11e6-822b-0242ac110004","message":"SKIPPED pgporada.vim : Ensure .vimrc exists","type":"ansible","ansible_playbook":"/tmp/kitchen/default.yml","ansible_task":"pgporada.vim : Ensure .vimrc exists","tags":[],"path":"/tmp/kitchen/callback_plugins/logstash.py","ansible_host":"localhost","@timestamp":"2017-02-14T06:08:28.147Z","port":49750,"host":"3c80185cc093","@version":"1","logger_name":"python-logstash-logger","status":"SKIPPED"}}
{"_index":"logstash-2017.02.14","_type":"ansible","_id":"AVo7PJA-vGV8lhZBQyFS","_score":1,"_source":{"ansible_type":"finish","level":"INFO","ansible_result":"{\"localhost\": {\"unreachable\": 0, \"skipped\": 3, \"ok\": 8, \"changed\": 0, \"failures\": 0}}","session":"ff0b955e-f27b-11e6-822b-0242ac110004","message":"{\"localhost\": {\"unreachable\": 0, \"skipped\": 3, \"ok\": 8, \"changed\": 0, \"failures\": 0}}","type":"ansible","ansible_playbook":"/tmp/kitchen/default.yml","tags":[],"path":"/tmp/kitchen/callback_plugins/logstash.py","@timestamp":"2017-02-14T06:08:28.940Z","port":49750,"@version":"1","host":"3c80185cc093","logger_name":"python-logstash-logger","status":"OK"}}
{"_index":"logstash-2017.02.14","_type":"ansible","_id":"AVo7OmGLvGV8lhZBQyFK","_score":1,"_source":{"ansible_type":"task","level":"INFO","ansible_result":"{\"ansible_facts\": {\"__vim_package\": \"vim\"}, \"changed\": false}","session":"ab28360e-f27b-11e6-b8c1-0242ac110004","ansible_changed":false,"message":"{\"ansible_facts\": {\"__vim_package\": \"vim\"}, \"changed\": false}","type":"ansible","ansible_playbook":"/tmp/kitchen/default.yml","ansible_task":"pgporada.vim : Include OS family specific variables","tags":[],"path":"/tmp/kitchen/callback_plugins/logstash.py","ansible_host":"localhost","@timestamp":"2017-02-14T06:06:05.312Z","port":49716,"host":"3c80185cc093","@version":"1","logger_name":"python-logstash-logger","status":"OK"}}
{"_index":"logstash-2017.02.14","_type":"ansible","_id":"AVo7OmHJU0QLp7y3Mj6a","_score":1,"_source":{"ansible_type":"task","level":"INFO","session":"ab28360e-f27b-11e6-b8c1-0242ac110004","message":"SKIPPED pgporada.vim : Apt-get update","type":"ansible","ansible_playbook":"/tmp/kitchen/default.yml","ansible_task":"pgporada.vim : Apt-get update","tags":[],"path":"/tmp/kitchen/callback_plugins/logstash.py","ansible_host":"localhost","@timestamp":"2017-02-14T06:06:05.347Z","port":49716,"host":"3c80185cc093","@version":"1","logger_name":"python-logstash-logger","status":"SKIPPED"}}
```

**Comment by pgporada** [comment]
`python-demjson`
```
$ jsonlint-py logstash-2017.02.14.data.json 
logstash-2017.02.14.data.json:2:0: Error: Unexpected text after end of JSON value
   |  At line 2, column 0, offset 500
logstash-2017.02.14.data.json: has errors
```

`npm install jsonlint`
```
$ jsonlint logstash-2017.02.14.data.json 
[Error: Parse error on line 1:
...er","status":"OK"}}{"_index":"logstash-
----------------------^
Expecting 'EOF', '}', ',', ']', got '{']
```

`json`
```
$ json validate < logstash-2017.02.14.data.json 
Traceback (most recent call last):
  File "/usr/local/bin/json", line 11, in <module>
    sys.exit(main())
  File "/usr/local/lib/python2.7/dist-packages/jsonspec/cli.py", line 541, in main
    args.func(args)
  File "/usr/local/lib/python2.7/dist-packages/jsonspec/cli.py", line 205, in __call__
    return self.run(args)
  File "/usr/local/lib/python2.7/dist-packages/jsonspec/cli.py", line 477, in run
    parse_document(args)
  File "/usr/local/lib/python2.7/dist-packages/jsonspec/cli.py", line 150, in parse_document
    document = driver.load(sys.stdin)
  File "/usr/local/lib/python2.7/dist-packages/jsonspec/driver.py", line 23, in load
    return _load(fp, *args, **kwargs)
  File "/usr/lib/python2.7/json/__init__.py", line 291, in load
    **kw)
  File "/usr/lib/python2.7/json/__init__.py", line 339, in loads
    return _default_decoder.decode(s)
  File "/usr/lib/python2.7/json/decoder.py", line 367, in decode
    raise ValueError(errmsg("Extra data", s, end, len(s)))
ValueError: Extra data: line 2 column 1 - line 18 column 1 (char 500 - 59400)
```

**Comment by evantahler** [comment]
This tool doesn't intend for the dumpfiles to be valid JSON.  In fact, they used to be, and that was changed explicitly in version 1.0 https://github.com/taskrabbit/elasticsearch-dump/releases/tag/v1.0.0.  The problem was that the data wasn't streamable, nor append-able. 

The data format is "json objects, one-per-line".  Back when 

Perhaps the README is misleading, calling the output files `*.json`... do you have a better suggestion?

**Comment by pgporada** [comment]
@evantahler Thanks for getting back to me! I posted out of frustration late last night without doing additional testing on my end. I've found that I can successfully validate the output from elasticdump by reading the output line by line into something like `jsonlint` or `json`.

```
while read LINE; do jsonlint-py "${LINE}" ; done < logstash-2017.02.14.data.json
```

An update to the readme explicitly stating that a person needs to read in the files line by line would be very helpful!
